File: benchmarks/fixtures/sqli.php
<?php

class LookupModel {
    public function vulnerableLookup($order_id) {
        return $this->db->query("SELECT * FROM `orders` WHERE `order_id` = " . $order_id);
    }

    public function numericLookup($order_id) {
        return $this->db->query("SELECT * FROM `orders` WHERE `order_id` = " . (int)$order_id);
    }
}

function vulnerableHaving($filter) {
    $sql_having = ' HAVING (';
    $sql_having .= 'name LIKE "%' . $filter . '%"';
    $sql_having .= ')';
    return $sql_having;
}

function quotedHaving($filter) {
    $sql_having = ' HAVING (';
    $sql_having .= 'name LIKE ' . db_qstr('%' . $filter . '%');
    $sql_having .= ')';
    return $sql_having;
}

File: benchmarks/fixtures/sqli.py
def vulnerable_lookup(request, cursor):
    query = request.GET.get("query", "")
    sql = "SELECT value FROM values WHERE LOWER(value)='{query}'"
    sql = sql.format(query=query.lower())
    cursor.execute(sql)


def parameterized_lookup(request, cursor):
    cursor.execute(
        "SELECT value FROM values WHERE LOWER(value)=%(query)s",
        {"query": request.GET.get("query", "").lower()},
    )
